eng/targets/GeneratePkgDef.targets

```diff
@@ -0,0 +1,221 @@
+          Inputs="$(MSBuildAllProjects)"
+          Outputs="$(_GeneratePkgDefOutputFile)"
+          Condition="'@(PkgDefEntry)' != ''">
+    <ItemGroup>
+      <_AssemblyDependencyEntry Include="@(PkgDefEntry)" Condition="'%(PkgDefEntry.Kind)' == 'BindingRedirect' or '%(PkgDefEntry.Kind)' == 'CodeBase'" />
+        
+      <_AssemblyDependencyEntryWithoutAssemblyName Include="@(_AssemblyDependencyEntry)" Condition="'%(_AssemblyDependencyEntry.FusionName)' == ''" />
+    </ItemGroup>
+    
+     <Microsoft.DotNet.Arcade.Sdk.GetAssemblyFullName Items="@(_AssemblyDependencyEntryWithoutAssemblyName)" FullNameMetadata="FusionName">
+       <Output TaskParameter="ItemsWithFullName" ItemName="_AssemblyDependencyEntryWithAssemblyName"/>
+     </Microsoft.DotNet.Arcade.Sdk.GetAssemblyFullName>
+  
+    <ItemGroup>
+      <_AssemblyDependencyEntryWithAssemblyName Include="@(_AssemblyDependencyEntry)" Condition="'%(_AssemblyDependencyEntry.FusionName)' != ''"/>
+      
+      <_AssemblyDependencyEntryWithAssemblyName Update="@(_AssemblyDependencyEntryWithAssemblyName)">
+        <AssemblyFileName>%(_AssemblyDependencyEntryWithAssemblyName.FileName)%(_AssemblyDependencyEntryWithAssemblyName.Extension)</AssemblyFileName>
+        <OldAssemblyVersion>0.0.0.0</OldAssemblyVersion>
+       
+        <!-- Split FusionName: "{0}, Version={1}, Culture={2}, PublicKeyToken={3}" -->
+        <AssemblyName>$([MSBuild]::ValueOrDefault('%(_AssemblyDependencyEntryWithAssemblyName.FusionName)', '').Split(',')[0])</AssemblyName>
+        <AssemblyVersion>$([MSBuild]::ValueOrDefault('%(_AssemblyDependencyEntryWithAssemblyName.FusionName)', '').Split(',')[1].Split('=')[1])</AssemblyVersion>
+        <Culture>$([MSBuild]::ValueOrDefault('%(_AssemblyDependencyEntryWithAssemblyName.FusionName)', '').Split(',')[2].Split('=')[1])</Culture>
+        <PublicKeyToken>$([MSBuild]::ValueOrDefault('%(_AssemblyDependencyEntryWithAssemblyName.FusionName)', '').Split(',')[3].Split('=')[1])</PublicKeyToken>
+      </_AssemblyDependencyEntryWithAssemblyName>
+
+      <_AssemblyDependencyEntryWithAssemblyName Update="@(_AssemblyDependencyEntryWithAssemblyName)" Condition="'%(PkgDefEntry)' == 'BindingRedirect'">
+        <HashBase>%(AssemblyName),%(PublicKeyToken),%(Culture),%(OldAssemblyVersion)-%(AssemblyVersion),%(AssemblyVersion)</HashBase>
+      </_AssemblyDependencyEntryWithAssemblyName>
+      
+      <_AssemblyDependencyEntryWithAssemblyName Update="@(_AssemblyDependencyEntryWithAssemblyName)" Condition="'%(PkgDefEntry)' == 'CodeBase'">
+        <HashBase>%(AssemblyName),%(PublicKeyToken),%(Culture),%(AssemblyVersion)</HashBase>
+      </_AssemblyDependencyEntryWithAssemblyName>
+    </ItemGroup>
+    
+    <Microsoft.DotNet.Build.Tasks.VisualStudio.GetPkgDefAssemblyDependencyGuid Items="@(_AssemblyDependencyEntryWithAssemblyName)" InputMetadata="HashBase" OutputMetadata="Guid">
+      <Output TaskParameter="OutputItems" ItemName="_ADEntry"/>
```

Comment:
I admit having both _AssemblyDependencyEntry and _ADEntry feels like a bug waiting to happen...

Reply:
I run out of names :( _AssemblyDependencyEntryWithAllMetadata is too long to work with

---
In reply to: [408487010](https://github.com/dotnet/roslyn/pull/43302#discussion_r408487010) [](ancestors = 408487010)